internal/modelcontroller/model_source.go

```diff
@@ -214,28 +215,42 @@ func (r *ModelReconciler) pvcPodAdditions(url modelURL) *modelSourcePodAdditions
 	}
 }
 
-var modelURLRegex = regexp.MustCompile(`^([a-z]+):\/\/(\S+)$`)
+var modelURLRegex = regexp.MustCompile(`^([a-z]+):\/\/([^?]+)(\?.*)?$`)
 
 func parseModelURL(urlStr string) (modelURL, error) {
 	matches := modelURLRegex.FindStringSubmatch(urlStr)
-	if len(matches) != 3 {
+	if len(matches) != 3 && len(matches) != 4 {
 		return modelURL{}, fmt.Errorf("invalid model URL: %s", urlStr)
 	}
 	scheme, ref := matches[1], matches[2]
 	name, path, _ := strings.Cut(ref, "/")
+	var modelParam string
+
+	if len(matches) == 4 { // check for query parameter model, e.g. pvc://my-pvc?model=qwen2:0.5b
+		urlParser, err := url.ParseQuery(strings.TrimPrefix(matches[3], "?"))
+		if err == nil {
+			modelname := urlParser.Get("model")
+			if modelname != "" {
+				modelParam = modelname
+			}
+		}
+	}
+
 	return modelURL{
-		original: urlStr,
-		scheme:   scheme,
-		ref:      ref,
-		name:     name,
-		path:     path,
+		original:   urlStr,
+		scheme:     scheme,
+		ref:        ref,
+		name:       name,
+		path:       path,
+		modelParam: modelParam,
 	}, nil
 }
 
 type modelURL struct {
-	original string // e.g. "hf://username/model"
-	scheme   string // e.g. "hf", "s3", "gs", "oss"
-	ref      string // e.g. "username/model"
-	name     string // e.g. username or bucket-name
-	path     string // e.g. model or path/to/model
+	original   string // e.g. "hf://username/model"
+	scheme     string // e.g. "hf", "s3", "gs", "oss", "pvc"
+	ref        string // e.g. "username/model"
+	name       string // e.g. username or bucket-name
+	path       string // e.g. model or path/to/model
+	modelParam string // e.g. "qwen2:0.5b when ?model=qwen2:0.5b is part of the URL
```

Comment:
Add a comment explaining *why* the `modelParam` field exists... its b/c in the use case of PVCs with ollama you end up with a multi-model directory structure and you need to be specific as to which one to load... that will likely be lost on a future reader.

Reply:
done